File: python/crypto-square/crypto_square.py
import re
import math
def cipher_text(plain_text):
    cleaned = re.sub(r"\W", "", plain_text).lower()
    c = int(math.sqrt(len(cleaned)))
    if c == 0: return ''
    if c ** 2 < len(cleaned): c = c + 1
    groups = [cleaned[i:i+c] for i in range(0, len(cleaned), c)]
    if not groups: return ''
    groups[-1] = groups[-1].ljust(c, " ")
    inverted = [''.join(l) for l in zip(*groups)]
    return ' '.join(inverted)

File: ruby/crypto-square/crypto_square.rb
class Crypto

  def initialize(input)
    @input = input
  end

  def ciphertext

    text = @input.gsub(/\W/,"").downcase
    l = text.length
    return "" if l < 1

    c = Math.sqrt(l)
    r = (l / c).ceil
    c = c.round

    text.ljust(c * r, padstr=" ")
        .chars
        .each_slice(r)
        .to_a
        .transpose
        .map(&:join)
        .join(" ")
  end
end
